internal/pkg/client/request.go

```diff
@@ -177,7 +177,12 @@ func (r *Request) WaitFor(subRequest *Request) {
 	// Continue execution of the listeners, when the sub-request completes
 	r.waitingForGrp.Add(1)
 	subRequest.OnResponse(func(response *Response) {
-		r.waitingForGrp.Done()
+		defer r.waitingForGrp.Done()
+
+		// Copy error from sub-request to parent request
+		if subRequest.HasError() && !r.HasError() {
```

Comment:
Včera malo KBC vypadol ... a CLI error handling nefungoval uplne podla predstav.

- Operacie `create branch`/`delete branch` su async, spustaju storage job.
- Tento job sa musi kontrolovat opakovane, pokial je v stave `waiting`.
- Existuje funkcia `remote.waitForJob`, ktora to robi.
- Je tam retry s nejakym limitom.
- Problem nastal ked tento sub-request, ktory kontroloval job skoncil chybov.
- Chyba sa neprepisala do hlavneho requestu, a tak program pokracoval dalej.
- Je to preto, lebo `create/delete branch` nespustame v `pool` ale ako synchronne requesty.
- Tj. s `pool` by spadlo `pool`, kedze jeden request nepresie,  ... ale bez `pool` sa pokracovalo.
- Preto TU kopirujeme chybu z sub-requestu.